developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,246 @@
+import re
+import subprocess
+
+from developers_chamber.scripts import cli
+from developers_chamber.utils import print_bold, print_error, print_heading, print_success
+from git import Repo
+
+
+class QAError(Exception):
+    def __init__(self, msg, output):
+        super().__init__(msg)
+        self.output = output.strip()
+
+
+class QACheck:
+    """
+    Base class for a quality assurance check.
+
+    Arguments:
+        name: Name of the check.
+    """
+    name = None
+
+    def _get_repo(self):
+        """
+        Returns the repo object.
+        """
+        return Repo('.')
+
+    def _get_default_branch(self):
+        """
+        Returns default branch of the repo.
+        """
+        return self._run_command("git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'")
+
+    def _get_diffs(self):
+        """
+        Returns Git diffs against default branch.
+        """
+        repo = self._get_repo()
+        target_commit = repo.heads[self._get_default_branch()].commit
+        return target_commit.diff('HEAD', create_patch=True)
+
+    def _get_unstaged(self):
+        """
+        Returns unstaged files in the repo.
+        """
+        return self._get_repo().index.diff(None, create_patch=True)
+
+    def _run_command(self, command):
+        """
+        Runs shell command and returns its output.
+        """
+        try:
+            return subprocess.check_output(command, shell=True).decode().strip()
```

Comment:
There's already call_command function in utils.py. Are you sure you need this?

Reply:
I'll check it out and modify it if needed.

Reply:
I think my version is sorta different in the way I am capturing the output and also how I handle errors. I think I'll keep this version.